front/pages/api/w/[wId]/files/[fileId].ts

```diff
@@ -18,8 +19,33 @@ export const config = {
   },
 };
 
-const validActions = ["view", "download"] as const;
-type Action = (typeof validActions)[number];
+// Declared here because endpoint-specific.
+//TODO: Create an enum for this class
```

Comment:
FWIW we don't use the `enum` type in Typescript.

Reply:
Why not ? This seems to be a good use case, because we hardcode these 'original', 'processed', ... in multiple places

Reply:
@overmode we use type options = "optionA" | "optionB" etc...

Reply:
Here's why we avoid enums in TS:
- numeric values by default: enums automatically use numbers unless specified otherwise, which can lead to unexpected serialized values in your database
- value shifting: adding a new value in between existing enum values changes all the subsequent numeric values, potentially breaking stored data
- union types are more recent: enums were added before union types existed in TS. Union types (A | B) are now the more modern and type-safe approach

Instead we use:
```
const VALUES = ["a", "b", "c"] as const;
type Value = (typeof VALUES)[number];
```

Reply:
Got it, is there any reason why you'd use this instead of an enum?